manifests/init.pp

```diff
@@ -1,16 +1,33 @@
 class clamps (
-  $num_dynamic_files = 5,
-  $num_static_files = 20,
+  $num_dynamic_files     = 5,
+  $num_static_files      = 20,
+  $num_static_files_step = 5,
 ) {
 
   file { [
       "/home/${id}/clamps_files/",
       "/home/${id}/clamps_files/static/",
       "/home/${id}/clamps_files/dynamic/",
+      "/home/${id}/.facter",
+      "/home/${id}/.facter/facts.d"
     ]:
     ensure => directory,
   }
 
+  $static_files_fact_path = "/home/${id}/.facter/facts.d/static_files.txt"
+
+  exec { 'create static_files fact' :
+    command => "/bin/echo static_files=50 > ${static_files_fact_path}",
+    creates => $static_files_fact_path,
+  }
+
+  $num_static_files_to_make_now = min( pick($::static_files, 0) + $num_static_files_step, $num_static_files)
+
+  file { $static_files_fact_path :
+    ensure  => present,
+    content => "static_files=${num_static_files_to_make_now}",
+  }
+
```

Comment:
@rick yes your description is correct.  Upon further thought, I'm not sure the exec statement is necessary... at the time I probably thought I needed the file to exist to access the fact but then I added the `pick($::static_files, 0)`... 

Let me try removing the exec statement as I suspect it's unnecessary.  Especially since the code I provided means it would be created with 50 and then immediately reduced to 5.